java/vector/src/main/java/org/apache/arrow/vector/complex/impl/StructOrListWriterImpl.java

```diff
@@ -79,7 +80,7 @@ public void end() {
    * Creates a new writer for a struct with the given name.
    */
   public StructOrListWriter struct(final String name) {
-    assert struct != null;
+    Preconditions.checkNotNull(struct);
```

Comment:
in this file, I think it is probably ok to simply remove assets or rever them.  in each case they will NPE on the next line.